org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/nogrammarconstraints/NoGrammarConstraintsCodeAction.java

```diff
@@ -246,18 +249,32 @@ public static TextDocumentEdit createXSINoNamespaceSchemaLocationEdit(String sch
 			throws BadLocationException {
 		String delimiter = document.getTextDocument().lineDelimiter(0);
 		DOMElement documentElement = document.getDocumentElement();
-		int beforeTagOffset = documentElement.getStartTagOpenOffset();
-		int afterTagOffset = beforeTagOffset + 1 + documentElement.getTagName().length();
+		int beforeTagOffset = documentElement != null ? documentElement.getStartTagOpenOffset() : document.getLastChild() != null ? document.getLastChild().getEnd() : 0;
+		int afterTagOffset = documentElement != null ? beforeTagOffset + 1 + documentElement.getTagName().length() : beforeTagOffset;
 
 		StringBuilder insertText = new StringBuilder();
 
+		if (documentElement == null) {

```

Comment:
Move this code in a private static method like `generateStartTag(DOMElement documentElement) `to reuse it with other create*Edit